File: 0048-rotate-image/0048-rotate-image.java
class Solution {
    public void rotate(int[][] arr) {
        int row_leng = arr.length;
        int col_leng = arr[0].length;

        // Transpose of matrix
        for(int i = 0; i < row_leng; i++){
            for (int j = i; j < col_leng; j++){
                int temp = arr[i][j];
                arr[i][j] = arr[j][i];
                arr[j][i] = temp;
            }
        }
        // swap each row 
        for(int i = 0; i < row_leng; i++ ){
            for(int j =0; j < (col_leng/2); j++){
                int temp = arr[i][j];
                arr[i][j]= arr[i][col_leng-1-j];
                arr[i][col_leng-1-j] = temp;
            }
        }

        
    }
}

File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    void transpose(vector<vector<int>>& matrix){
        int row = matrix.size();
        for(int i=0; i<row; i++){
            for(int j=i; j<row; j++ ){
                swap(matrix[i][j],matrix[j][i]);
            }
        }
    }

    void rev(vector<vector<int>> &matrix){
        int row = matrix.size();
        int col = matrix[0].size();
        
        for(int i=0; i<row; i++){
            int low = 0;
            int high = row-1;

            while(low<=high){
                swap(matrix[i][low],matrix[i][high]);
                low++;
                high--;
            }
        }

    }
    void rotate(vector<vector<int>>& matrix) {
        transpose(matrix);
        rev(matrix);
        
    }
};